pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -164,13 +143,14 @@ func newManagerImpl(socketPath string, topology []cadvisorapi.Node, topologyAffi
 		topologyAffinityStore: topologyAffinityStore,
 		devicesToReuse:        make(PodReusableDevices),
 	}
-	manager.callback = manager.genericDeviceUpdateCallback
+	manager.server = plugin.NewServer(socketPath, manager, manager)
+	manager.checkpointdir, _ = filepath.Split(socketPath)
```

Comment:
minor: since we're doing refactoring, I wonder if we should add a function to extract the checkpointdir from the socketPath (or perhaps even from a Server?) and abstract away this detail.

Reply:
I couldn't find an obvious way to do this. If you have a suggestion, I'm happy to hear it.